recipe-client-addon/test/unit/test_PreferenceExperiments.js

```diff
@@ -110,6 +165,25 @@ add_task(withMockPreferences(function* (mockPreferences) {
   stop.restore();
 }));
 
+// startObserver should observe changes to the default preference value.
+add_task(withMockPreferences(function* (mockPreferences) {
+  const stop = sinon.stub(PreferenceExperiments, "stop");
+  mockPreferences.set("fake.preference", "startvalue", "default");
+
+  PreferenceExperiments.startObserver("test", "fake.preference", "experimentvalue");
+
+  // Setting it to the experimental value should not trigger the call.
+  DefaultPreferences.set("fake.preference", "experimentvalue");
+  ok(!stop.called, "Changing to the experimental pref value did not trigger the observer");
+
+  // Setting it to something different should trigger the call.
```

Comment:
Not entirely sure what we're testing here, and if it's worth testing this again once setting it back to the experiment value from the new value? Specifically, is calling `startObserver` expected to set the preference immediately, or is it not set until the test here calls `DefaultPreferences.set` ? In the latter case, I suppose testing this again would make no difference, but a comment might be nice. Otherwise, testing the correct behaviour (whatever that is) when setting the thing back to the expected experiment value would make sense.